src/main/java/gregtech/api/items/metaitem/stats/ItemFluidContainer.java

```diff
@@ -0,0 +1,21 @@
+package gregtech.api.items.metaitem.stats;
+
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandlerItem;
+
+public class ItemFluidContainer implements IItemContainerItemProvider {
+    @Override
+    public ItemStack getContainerItem(ItemStack itemStack) {
+        IFluidHandlerItem handler = itemStack.getCapability(CapabilityFluidHandler.FLUID_HANDLER_ITEM_CAPABILITY, null);
+        if (handler != null) {
+            FluidStack drained = handler.drain(1000, false);
+            if (drained == null) return ItemStack.EMPTY;
+            if (drained.amount == 1000) {
```

Comment:
You have added this to cells that can contain greater than 1000 mb of fluid, as well as contain variable amounts of fluid, maybe being less than 1000 mb. Should this instead check if `drained.amount > 0`? and should drained be changed to `handler.drain(Integer.MAX_VALUE, false)`?